### PR TITLE
feat(DEVOPS-8075): limit HTTP messages size by nginx

### PR DESCRIPTION
**Pull request by christophemorio**
Jira: [DEVOPS-8075](https://jira.talendforge.org/browse/DEVOPS-8075)

### Expected
For AWS, installation of Nginx will be apply via talend-cloud-installer.

Adapt the existing temporary Ansible https://github.com/Talend/tmp-nginx-amq to puppet code.
Adapt the existing Terraform code to reflect temporary changes.

### Actions
- [x] Append to the existing *puppet role* activemq, a new *puppet profile* nginx_reverseproxy
- [ ] Update LoadBalancer

### CODE REVIEW

File: hieradata/environment/vagrant.yaml [hunk 1]
```diff
@@ -159,3 +159,4 @@ profile::docker::registry::registries:
     password: 'my_password'
 
 profile::nexus::nexus_mem_check::enabled: true
+profile::nginx_reverseproxy::client_max_body_size: 50m
```
Comment (wanix) [inline hieradata/environment/vagrant.yaml]: for local test, you can use 200k

File: site/profile/manifests/nginx_reverseproxy.pp [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#
+# Nginx service profile
+#
+# Setup a proxypass to ActiveMQ filtering messages by size
+#
+class profile::nginx_reverseproxy (
+
+  $client_max_body_size = '200k',
```
Comment (wanix) [inline site/profile/manifests/nginx_reverseproxy.pp]: real value should be more 50m

File: site/profile/manifests/nginx_reverseproxy.pp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    proxy_redirect          => 'off',
+    proxy_send_timeout      => '60s',
+    proxy_set_header        => [],
+    types_hash_max_size     => 2048,
+    worker_connections      => 8192,
+    worker_processes        => 'auto',
+    worker_rlimit_nofile    => 102400,
+  }
+
+  nginx::resource::vhost { 'jetty':
```
Comment (wanix) [inline site/profile/manifests/nginx_reverseproxy.pp]: What about access and errors logs (rotation on size, date, no logs, aws clouwatch agent ?)

File: site/profile/manifests/nginx_reverseproxy.pp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    proxy_buffer_size       => undef,
+    proxy_connect_timeout   => '60s',
+    proxy_read_timeout      => '60s',
+    proxy_redirect          => 'off',
+    proxy_send_timeout      => '60s',
+    proxy_set_header        => [],
+    types_hash_max_size     => 2048,
+    worker_connections      => 8192,
+    worker_processes        => 'auto',
+    worker_rlimit_nofile    => 102400,
```
Comment (wanix) [inline site/profile/manifests/nginx_reverseproxy.pp]: you should have less than 102400 there, 102000 ?

File: site/profile/manifests/nginx_reverseproxy.pp [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  class { 'nginx':
+    client_body_buffer_size => false,
+    client_max_body_size    => $client_max_body_size,
+    gzip                    => 'off',
+    http_tcp_nodelay        => 'on',
+    http_tcp_nopush         => 'on',
+    keepalive_timeout       => 65,
+    proxy_buffers           => undef,
+    proxy_buffer_size       => undef,
+    proxy_connect_timeout   => '60s',
```
Comment (wanix) [inline site/profile/manifests/nginx_reverseproxy.pp]: please verify 60s compared to current values for Jetty

File: site/role/manifests/activemq.pp [hunk 6]
```diff
@@ -5,6 +5,7 @@
 
   include ::profile::base
   include ::profile::activemq
+  include ::profile::nginx_reverseproxy
```
Comment (wanix) [inline site/role/manifests/activemq.pp]: file/class naming: nginx_activemq_reverse_proxy ?

File: spec/acceptance/shared/nginx_reverseproxy.rb [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    its(:exit_status) { should eq 0 }
+  end
+
+  describe 'Expects an error 500 by requesting 100KB payload' do
+    subject { command('/usr/bin/curl -s -o /dev/null -w "%{http_code}" --data-binary @/tmp/100K.dump http://localhost/') }
+    its(:stdout) { should eq '500' }
+  end
+
+  describe 'Expects an error 413 by requesting 100MB payload' do
+    subject { command('/usr/bin/curl -s -o /dev/null -w "%{http_code}" --data-binary @/tmp/100M.dump http://localhost/') }
```
Comment (wanix) [inline spec/acceptance/shared/nginx_reverseproxy.rb]: you can test with 300k only with a kitchen/vagrant configured with 200k limit

File: site/role/manifests/activemq.pp (hunk 6 above)
Comment (christophemorio) [reply site/role/manifests/activemq.pp]: actually in progress with `nginx_amq_reverseproxy`

File: site/profile/manifests/nginx_reverseproxy.pp (hunk 3 above)
Comment (christophemorio) [reply site/profile/manifests/nginx_reverseproxy.pp]: The Jira does not covers the needs for long term solution with monitoring.
Log rotation uses defaults settings from nginx.

**Pull request merged**